internal/riverinternaltest/riverdrivertest/riverdrivertest.go

```diff
@@ -0,0 +1,1118 @@
+
+			exec, _ := setupExecutor(ctx, t, driver, beginTx)
+
+			// Insert jobs with decreasing priority numbers (3, 2, 1) which means increasing priority.
+			for i := 3; i > 0; i-- {
+				_ = testfactory.Job(ctx, t, exec, &testfactory.JobOpts{
+					Priority: &i,
+				})
+			}
+
+			jobRows, err := exec.JobGetAvailable(ctx, &riverdriver.JobGetAvailableParams{
+				AttemptedBy: clientID,
+				Max:         2,
+				Queue:       rivercommon.QueueDefault,
+			})
+			require.NoError(t, err)
+			require.Len(t, jobRows, 2, "expected to fetch exactly 2 jobs")
+
+			// Because the jobs are ordered within the fetch query's CTE but *not* within
+			// the final query, the final result list may not actually be sorted. This is
+			// fine, because we've already ensured that we've fetched the jobs we wanted
+			// to fetch via that ORDER BY. For testing we'll need to sort the list after
+			// fetch to easily assert that the expected jobs are in it.
+			sort.Slice(jobRows, func(i, j int) bool { return jobRows[i].Priority < jobRows[j].Priority })
+
+			require.Equal(t, 1, jobRows[0].Priority, "expected first job to have priority 1")
+			require.Equal(t, 2, jobRows[1].Priority, "expected second job to have priority 2")
+
+			// Should fetch the one remaining job on the next attempt:
+			jobRows, err = exec.JobGetAvailable(ctx, &riverdriver.JobGetAvailableParams{
+				AttemptedBy: clientID,
+				Max:         1,
+				Queue:       rivercommon.QueueDefault,
+			})
+			require.NoError(t, err)
+			require.NoError(t, err)
+			require.Len(t, jobRows, 1, "expected to fetch exactly 1 job")
+			require.Equal(t, 3, jobRows[0].Priority, "expected final job to have priority 3")
+		})
+	})
+
+	t.Run("JobGetByID", func(t *testing.T) {
+		t.Parallel()
+
+		t.Run("FetchesAnExistingJob", func(t *testing.T) {
+			t.Parallel()
+
+			exec, _ := setupExecutor(ctx, t, driver, beginTx)
+
+			now := time.Now().UTC()
+
+			job := testfactory.Job(ctx, t, exec, &testfactory.JobOpts{})
+
+			job, err := exec.JobGetByID(ctx, job.ID)
+			require.NoError(t, err)
+			require.NotNil(t, job)
+
+			require.Equal(t, job.ID, job.ID)
+			require.Equal(t, rivertype.JobStateAvailable, job.State)
+			require.WithinDuration(t, now, job.CreatedAt, 100*time.Millisecond)
+			require.WithinDuration(t, now, job.ScheduledAt, 100*time.Millisecond)
+		})
+
+		t.Run("ReturnsErrNoRowsIfJobDoesntExist", func(t *testing.T) {
+			t.Parallel()
+
+			exec, _ := setupExecutor(ctx, t, driver, beginTx)
+
+			job, err := exec.JobGetByID(ctx, 99999)
+			require.Error(t, err)
+			require.ErrorIs(t, err, rivertype.ErrNotFound)
+			require.Nil(t, job)
+		})
+	})
+
+	t.Run("JobGetByIDMany", func(t *testing.T) {
+		t.Parallel()
+	})
+
+	t.Run("JobGetByKindAndUniqueProperties", func(t *testing.T) {
+		t.Parallel()
+	})
+
+	t.Run("JobGetByKindMany", func(t *testing.T) {
+		t.Parallel()
+	})
+
+	t.Run("JobGetStuck", func(t *testing.T) {
+		t.Parallel()
+	})
+
+	t.Run("JobInsert", func(t *testing.T) {
+		t.Parallel()
+	})
```

Comment:
Are these still meant to be empty? same comment for the other empty ones in here

Reply:
Yep, caught me. These were for functions where there were no equivalents from `adapter_test.go`, and I'd refactored so many tests already that I'd left these for an exercise for the future.

That said, had some time this evening, so I filled them all in. The test bootstrap here should actually be pretty good for testing a theoretical future driver now, even without having to put it in a real client. I also found a couple things that could be considered bugs, like some timestamps not being returned back as `UTC()` when they probably should've been.

You might want to go through eventually and double-check that you're happy with all the conditions being checked. A few could probably use a couple more cases.